File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int i =0 ;
        int j = nums.size()-1;
        int mini = INT_MAX;

        while(i<=j){
            int mid = (i+j)/2;
            if(nums[mid]>=nums[i]){
                mini = min(mini,nums[i]);
                i = mid+1;
            }
            else if(nums[j]>=nums[mid]){
                mini = min(mini,nums[mid]);
                j = mid-1;
            }
        }        
        return mini;
    }
};

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left , right = 0 , len(nums) -1 
        mini = float('inf')

        while(left <= right):
            mid = (left + right) // 2

            if (nums[left] <= nums[mid]):
                mini = min(mini,nums[left])
                left = mid + 1
            elif(nums[mid] <= nums[right]):
                mini = min(mini,nums[mid])
                right = mid - 1

        return mini